### DISCUSSION TITLE
Routes with no Oifs appearing.

### DISCUSSION THREAD

**Discussion by ghost** [troglobit/smcroute#196] [General]
Hi all! 

We have a use case along the lines of:
```
+-------------+        +-------------+           +------------+
| Client      |        | Server B    |           | Server A   |
|             |    eth1| 20.04       |eth0       |            |
| Proc Y      +--------+             +-----------+ Proc X     |
|             |        | smcroute    |           |            |
|             |        | 2.4.2-4     |           |            |
|             |        |             |           |            |
+-------------+        +-------------+           +------------+
 Proc Y                 Proc A                    Prox X
                        Proc B
```
Processes A. B, Z and Y all send and receive mcast. Server B runs smcroute and currently (for some code legacy reasons) the rules are installed with direct calls to `smcroute`, eg:
```
smcroute -a eth0 192.168.1.1 239.255.0.1 eth1
smcroute -a eth1 192.168.1.1 239.255.0.1 eth0
```
With the script containing them invoked by placing it in systemd's `/etc/networkd-dispatcher/routable.d` folder.

There's an issue that happens every 4 or 5 boots of Server B, that brings the route up with no `Oifs` on some of the routes, eg:
```
(192.168.1.1,239.255.0.1)    Iif: eth0  State: resolved
```
Here, `Client` behind `eth` doesn't receive the group messages. 
Running `sudo systemctl restart smcroute` and then re-running the above script resolves this:
```
(192.168.1.1,239.255.0.1)     Iif: eth0 Oifs: eth1  State: resolved
```
...and traffic flows

My question is: Is my script racing some other part of the system by trying to run via the `routable` interface hook? Or should this work and we're just hitting a bug that has perhaps been fixed in [2.4.3](https://github.com/troglobit/smcroute/blob/master/ChangeLog.md) (I'm not sure if the fix detailed there is for this issue; All the IPs in our stack are static)

Thanks!

**Comment by troglobit** [comment]
If you are running a version < 2.4.3, I suggest you upgrade to the latest version.  One particular bug you might run into is an old requirement that interfaces needed to have an IP address before SMCRoute started.

**Reply by ghost** [reply]
I wasn't wholly sure that the bug was related to what we were seeing - on the assumption that an interface wouldn't be routable if didn't yet have an IP - but perhaps there's nuance to this that I'm missing. This is systemd we're dealing with after all :) 

I'll need to build from source since there's no backport available for ubuntu 20.04, but I'll follow up once I've given that a go.

**Reply by troglobit** [reply]
routable != multicast routable.

**Reply by ghost** [reply]
No argument from me on that front :)  This is more about what _systemd_ regards as "routable" and hence where our script is being triggered. 

We've backported 2.5.5-1 from 22.04 and it's running in a test environment. Will update here once we've put it through its paces. If nothing else the addition of `/etc/smcroute.d` support is going to really help with our deployment + packaging.

**Reply by troglobit** [reply]
Great to hear, good luck!

**Reply by ghost** [reply]
Hi @troglobit - Update as promised:  We have rolled out the backported 2.5.5 package across most of our `Server B`s and this issue has evaporated. 

Thanks again for your help :)

**Reply by troglobit** [reply]
This is great news, thank you for reporting back! :-)
